src/backend/gog/games.ts

```diff
@@ -159,7 +159,14 @@ class GOGGame extends Game {
     const etaMatch = data.match(/ETA: (\d\d:\d\d:\d\d)/m)
     const bytesMatch = data.match(/Downloaded: (\S+) MiB/m)
     const progressMatch = data.match(/Progress: (\d+\.\d+) /m)
-    if (etaMatch && bytesMatch && progressMatch) {
+    if (
+      etaMatch &&
```

Comment:
If the array gets longer you could do:
```typescript
!etaMatch.some((entry) => entry === null))
```

Reply:
Not sure if that would help here, since I have to check that
1. `etaMatch` itself isn't `null`
2. the group provided in the RegExp (`etaMatch[1]`) matched something
I believe both of those cases can happen independently, at least TS warned me about it

Reply:
Should you then not test `etaMatch && etaMatch[0] && etaMatch[1]`. I don't know if typescript arrays work like C arrays where `etaMatch` would be the pointer to `etaMatch[0]`

Reply:
As long as `null` isn't returned, `etaMatch[0]` will always be the matched text. See [the spec on `RegExp.exec`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/exec#return_value) (called by `String.match` internally)

Reply:
Ok yes makes sense. This is fine then, aslong as we don't do something like this:
```typescript
if(etaMatch && etaMatch[1] && etaMatch[2] && etaMatch[3] && ...)
```